schema/CBP.yaml

```diff
@@ -0,0 +1,56 @@
+$schema: http://json-schema.org/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_CBP/blob/master/schema/CBP.yaml
+title: CBP v1
+description: Schema for CBP configuration files
+type: object
+properties:
+  address:
+    description: IP address of CBP
+    type: string
+  port:
+    description: port of CBP
+    type: integer
+  mask1:
+    description: Mask 1 of CBP
+    type: object
+    properties:
+      name:
+        description: name of mask
+        type: string
+      rotation:
+        description: rotation of mask
+        type: number
```

Comment:
Please list the units of rotation in the description string.

I'm feeling a bit conflicted about the data layout. It's quite clear but also quite verbose. Two alternatives that you might consider (though you probably already did); both lose a bit of clarity but save a lot of trouble when creating the configurations:
* Separate name and rotation arrays, each of which must have 5 elements.
* A single array of arrays. The first array must have 5 elements, and each contained array must be a tuple of string, number (name, rotation).

In either case jsonschema can easily validate the shape of the data, so you are sure to get exactly what you asked for.

Reply:
You didn't respond to this so I don't know if you saw it. Here is a concrete suggestion. This is the complete schema. Note how much shorter it is than your current one, yet it adds some missing validation:

```
$schema: http://json-schema.org/draft-07/schema#
$id: https://github.com/lsst-ts/ts_CBP/blob/master/schema/CBP.yaml
title: CBP v1
description: Schema for CBP configuration files
type: object
properties:
  address:
    description: IP address of CBP
    type: string
  port:
    description: port of CBP
    type: integer
  masks:
    description: Name and rotation angle for the mask in each slot of the mask wheel
    type: array
    minItems: 5
    maxItems: 5
    items:
        - type: string
          description: Name of mask
        - type: number
          description: Rotation angle of mask (degree)
          minimum: 0
          maximum: 360
required: [address, port, masks]
additionalProperties: false
```
And here is sample data for that schema:
```
address: 127.0.0.1
port: 35
masks:
  - name of mask 1: 45.1
  - an even longer name for mask 2: 36.2
  - and so on for mask 3: 189
  - and mask 4: 98
  - the last mask: 0
```
Data will fail validation if:
* any field is missing
* any field is added
* there are not exactly 5 elements of mask data
* any mask rotation angle is < 0 or > 360 (assuming you want to make such a restriction; perhaps you don't mind accepting other angles or would prefer a different range)

Reply:
If you do keep your current schema please add the `required` and `additionalProperties` fields to it. This prevents data being silently ignored if you misspell a field name.

Reply:
One more suggestion: if you know of a good default value for the address and port, I suggest provide it (using a `default: ...` specification).

Reply:
I updated the schema to validate required properties. I'll take this schema as part of the cleanup ticket.